tests/unit/test_discovery_client.py

```diff
@@ -16,8 +16,20 @@
     "ni.measurementlink.measurement.v2.MeasurementService",
 ]
 
+_PROVIDED_ANNOTATIONS = {
```

Comment:
Provided annotations for what? You have a bunch of different test serviceconfigs and this seems to have merged values from all of them. What happens when they confict?

FYI, the "provided" in `_PROVIDED_MEASUREMENT_SERVICES` refers to the `providedInterfaces` key and may not be relevant here.

Reply:
_PROVIDED_ANNOTATIONS is similar to _PROVIDED_MEASUREMENT_SERVICES.. the test just use this to form the ServiceInfo and then checks with itself.   Both are used for the test only.  

The reason I have like a mix, is that I wanted to add more cases to this single test.

Reply:
I still think it's confusing to combine annotations for 6 versions of the same service into a single dictionary. Does the feature really work this way? You can add multiple .serviceconfig files for the same service and the discovery service merges their annotations?

Reply:
On this test we build the service_info, so there is no different versions of .serviceconfig  
The Annotations I added all could be part of a .serviceconfig. And is not supposed to look good, is supposed to be the json embedded representation.

Reply:
I guess I am not understanding the confusion here.  It is being used exactly as _PROVIDED_MEASUREMENT_SERVICES.. I know that refers to providedInterfaces. And my Annotations refers to the same..  it is a valid case.

Reply:
Okay, now I see. These strings match multiple `.serviceconfig` files, but that is a coincidence because you copy/pasted them. This test doesn't use the `.serviceconfig` files at all.

Please change these strings so they don't match the `.serviceconfig` files. Otherwise, someone making changes to one might assume both copies are related and have to be kept in sync.

Reply:
Yeap, that I can do.